extensions/amp-sidebar/0.1/test/test-amp-sidebar.js

```diff
@@ -726,3 +734,47 @@ describes.realWin(
     });
   }
 );
+
+describes.realWin(
+  'amp-sidebar component with runtime on',
+  {
+    amp: {
+      extensions: ['amp-sidebar:0.1'],
+      runtimeOn: true,
+    },
+  },
+  (env) => {
+    it('should allow default actions in email documents', async () => {
+      env.win.document.documentElement.setAttribute('amp4email', '');
+      const action = new ActionService(env.ampdoc, env.win.document);
+      env.sandbox.stub(Services, 'actionServiceForDoc').returns(action);
+
+      const element = createElementWithAttributes(
+        env.win.document,
+        'amp-sidebar',
+        {'layout': 'nodisplay'}
+      );
+      env.win.document.body.appendChild(element);
+      const spy = env.sandbox.spy();
+      element.enqueAction = spy;
+      element.getDefaultActionAlias = env.sandbox.stub().returns({'items': []});
+      await whenUpgradedToCustomElement(element);
+
+      ['open', 'close', 'toggle'].forEach((method) => {
+        const i = new ActionInvocation(
+          element,
+          method,
+          /* args */ null,
+          'source',
+          'caller',
+          'event',
+          ActionTrust.HIGH,
+          'tap',
+          element.tagName
+        );
+        action.invoke_(i);
```

Comment:
For an `amp-sidebar` unit test, typically you can stub and return a fake for a service dependency. It's ok to use a real object, but I think we should only be spying and verifying that ActionService's public API is called as expected. 

By executing `invoke_()` here, we're testing a non-public API of a dependency.

> The important part is that you're only testing the API of your class and not the API of its dependencies

`go/mocks#prefer-real-objects`

Reply:
Thanks for the pointer! I replaced `invoke_` with `execute` in all the unit tests for components.